consensus/safety-rules/src/remote.rs

```diff
@@ -56,17 +56,21 @@ impl<T: Payload> RemoteService<T> {
 }
 
 pub struct RemoteClient<T> {
-    service: Arc<RwLock<RemoteService<T>>>,
+    service: Box<dyn TRemoteClient<T>>,
 }
 
 impl<T: Payload> RemoteClient<T> {
-    pub fn new(service: Arc<RwLock<RemoteService<T>>>) -> Self {
+    pub fn new(remote_service: Arc<RwLock<RemoteService<T>>>) -> Self {
+        let service = Box::new(LocalService { remote_service });
```

Comment:
i'm confused by those naming, `LocalService { remote_service}`

Reply:
Ahh, we should rename this service into SerializationService then the container could either be a Local (in memory) or Thread / Process for remote.

Reply:
renamed remote to serializer but did it in a new commit (appended to the end of this PR) as the Remote is in an earlier stack.